### PR TITLE
Content preview logging

### PR DESCRIPTION
**Pull request by JuliaFernee**
WIP contains 
healthcjecks
pull from master 9544975ad9deacbf3959090fd01af611aa45f04b committed 23 hours ago
some of logging

### CODE REVIEW

File: app.go [hunk 1]
```diff
@@ -2,109 +2,50 @@ package main
...
-	app := cli.App("content-preview", "A RESTful API for retrieving and transforming content preview data")
+
+	log.SetLevel(log.InfoLevel)
+	log.Infof("%s service started with args %s", serviceName, os.Args)
+
+	app := cli.App(serviceName, serviceDescription)
 	appPort := app.StringOpt("app-port", "8084", "Default port for app")
 	mapiAuth := app.StringOpt("mapi-auth", "default", "Basic auth for MAPI")
-	mapiUri := app.StringOpt("mapi-uri", "http://methode-api-uk-p.svc.ft.com/eom-file/", "Host and path for MAPI")
+	mapiUri := app.StringOpt("mapi-uri", "http://methode-api-uk-p.svc.ft.com", "Host and port for MAPI")
```
Comment (sjwells) [inline app.go:29]: The default value has no port - is that OK?

File: handlers.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+		mapiResp.Body.Close()
+		return
+	}
+	w.Header().Set(tid.TransactionIDHeader, matResp.Header.Get(tid.TransactionIDHeader))
+	io.Copy(w, matResp.Body)
+	matResp.Body.Close()
+}
+
+func ( h Handlers) getNativeContent(uuid string, transactionId string, w http.ResponseWriter) (ok bool, resp *http.Response) {
+	logger.Formatter = new(log.JSONFormatter)
```
Comment (sjwells) [inline handlers.go]: Go provides support for passing around variables associated with a request lifecycle via the context package, and 
recommended best practice for using this is to specify a context as the first argument to your function (see the [transaction id utils library](https://github.com/Financial-Times/transactionid-utils-go) for more info). Do you think we should use this here?

File: healthchecks.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package main
+import (
+	"net/http"
+	"fmt"
+	fthealth "github.com/Financial-Times/go-fthealth/v1a"
+)
+
+func (h Handlers) mapiCheck() fthealth.Check {
+	return fthealth.Check{
+		BusinessImpact:   "Articvle Preview Service will not work",
```
Comment (sjwells) [inline healthchecks.go]: 'Article' not 'Articvle' - and I would say instead something like 'Editorial users won't be able to preview articles as they will appear on the website'

File: healthchecks.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+		Name:             "Methode Api Availablilty Check",
```
Comment (sjwells) [inline healthchecks.go]: Availability not Availablilty (sorry, I am a proofreader!)

File: healthchecks.go [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+		PanicGuide:       "TODO - write panic guide",
+		Severity:         1,
+		TechnicalSummary: "Checks that Methode API Service is reachable. Article Preview Service requests native content from Methode API service.",
+		Checker:          func() (string, error) {
+			return checkServiceAvailablity("Methode API", h.mapiUri, h.mapiAuth)
```
Comment (sjwells) [inline healthchecks.go]: typo in this method name too

File: healthchecks.go [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+		},
+	}
+}
+
+func (h Handlers) matCheck() fthealth.Check {
+	return fthealth.Check {
+		BusinessImpact:   "Article Peview service will not work",
```
Comment (sjwells) [inline healthchecks.go]: Preview not Peview - and again, maybe rephrase?

File: healthchecks.go [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+		Name:             "Mehtode Article Transformer Availablilty Check",
```
Comment (sjwells) [inline healthchecks.go]: Methode not Mehtode, Availability not Availablilty

File: app.go [hunk 8]
```diff
@@ -2,109 +2,50 @@ package main
...
-
-func (h Handlers) contentPreviewHandler(w http.ResponseWriter, r *http.Request) {
-	log.Printf("received request");
-
-	vars := mux.Vars(r)
-	uuid := vars["uuid"]
-
-	if uuid == "" {
-		log.Fatal("missing UUID")
+		log.Fatal(http.ListenAndServe(":"+*appPort, nil))
```
Comment (sjwells) [inline app.go]: I suggest wrapping this in the metrics aware and transaction id aware http handlers - they will automatically capture metrics for each http method, and these can then be exported to graphite if you want them, and will also output a request log, that includes a transaction id.

example, where r is the mux router:

```
http.ListenAndServe(fmt.Sprintf(":%d", *port),
            httphandlers.HTTPMetricsHandler(metrics.DefaultRegistry,
                httphandlers.TransactionAwareRequestLoggingHandler(log.StandardLogger(), r)))
```

File: app.go (hunk 1 above)
Comment (JuliaFernee) [inline app.go:29]: It's ok

File: handlers.go (hunk 2 above)
Comment (JuliaFernee) [inline handlers.go]: I am planning to incorporate this approach in the next task when I add metrics

File: healthchecks.go (hunk 3 above)
Comment (JuliaFernee) [inline healthchecks.go]: +1

File: healthchecks.go (hunk 4 above)
Comment (JuliaFernee) [inline healthchecks.go]: +1

File: healthchecks.go (hunk 7 above)
Comment (JuliaFernee) [inline healthchecks.go]: +1
